### ISSUE TITLE
Validate scopes in token refresh?

### ISSUE THREAD

**Issue by jmandel** [onc-healthit/inferno#245]
Should the returned scopes in response to a token refresh be validated? Currently Inferno is okay with something like "oob" here:

```
{
  "access_token": "2c4cab52-67b6-4c42-981f-dfba700f66fb",
  "token_type": "Bearer",
  "expires_in": 3600,
  "refresh_token": "979638f9-646f-4d73-baf2-e20d7ec7ef04",
  "scope": "oob"
}
```

-- which is surprising; I think it'd be good to apply the same logic to compare these as in the initial token request.

**Comment by radamson** [comment]
Thanks for the report!

It does look like there needs to be more strict checking in the token refresh sequence so that the [scopes provided are a strict sub-set of the scopes granted in the original launch](http://www.hl7.org/fhir/smart-app-launch/index.html#step-5-later-app-uses-a-refresh-token-to-obtain-a-new-access-token).  I'll open a ticket for us to resolve this.

Right now Inferno is ok with a token like "oob" being returned in either authorization or refresh, but will provide a warning about not receiving the expected scopes.

Example when "oob" is provided as a scope:
![Screen Shot 2019-09-26 at 11 45 05 AM](https://user-images.githubusercontent.com/41651655/65703739-466d8200-e053-11e9-8029-33989bcad3bb.png)

Token refresh provides similar logic and warnings:

![Screen Shot 2019-09-26 at 11 47 15 AM](https://user-images.githubusercontent.com/41651655/65703844-7157d600-e053-11e9-9cd1-33ab4ca9b2a9.png)

**Comment by Jammjammjamm** [comment]
This was addressed in #380, but has not yet been deployed.

**Comment by radamson** [comment]
Addressed and deployed!  Feel free to reopen if further discussion is needed.
